src/app/pages/PerpetualPage/hooks/graphql/useGetTraderEvents.ts

```diff
@@ -28,19 +38,34 @@ function generateQuery(
   user: string,
   orderBy?: string,
   orderDirection?: OrderDirection,
+  page?: number,
+  perPage?: number,
 ): DocumentNode {
   const arr = events.map(event => {
     const eventDetails = EventDictionary.get(event);
+
+    const isOrdered = orderBy && orderDirection;
+    const hasPagination = page && perPage;
+
+    const orderFilterString = isOrdered
+      ? `orderBy: ${orderBy}, orderDirection: ${orderDirection}`
+      : '';
+
+    const paginationFilterString = hasPagination
+      ? `skip: ${(page! - 1) * perPage!} first: ${perPage!}`
+      : '';
+
     return `${eventDetails.entityName} ${
-      orderBy && orderDirection
-        ? `(orderBy: ${orderBy}, orderDirection: ${orderDirection})`
+      isOrdered || hasPagination
+        ? `(${orderFilterString} ${paginationFilterString})`
         : ''
     } { ${eventDetails.fields.toString()} }`;
   });
 
   return gql`
   { trader(id: "${user}")
   {
+    ${getTotalCountFieldName(events[0])}
```

Comment:
This should also map all event types in the array, and filter out unsupported ones.